src/storage/StorageConfig.js

```diff
@@ -0,0 +1,91 @@
+const fetch = require('node-fetch')
+
+const logger = require('../helpers/logger')('streamr:storage:StorageConfig')
+
+/*
+ * Connects to Core API and queries the configuration there.
+ * Refreshes the config at regular intervals.
+ */
+const getStreamFromKey = (key) => {
+    const [id, partitionStr] = key.split('::')
+    return {
+        id,
+        partition: Number(partitionStr)
+    }
+}
+
+const getKeyFromStream = (streamId, streamPartition) => {
+    return `${streamId}::${streamPartition}`
+}
+
+module.exports = class StorageConfig {
+    // use createInstance method instead: it fetches the up-to-date config from API
+    constructor(nodeId, apiUrl) {
+        this.streams = new Set()
+        this.listeners = []
+        this.nodeId = nodeId
+        this.apiUrl = apiUrl
+        this._poller = undefined
+        this._stopPoller = false
+    }
+
+    static async createInstance(nodeId, apiUrl, pollInterval = 10 * 60 * 1000) {
+        const instance = new StorageConfig(nodeId, apiUrl)
+        const poll = async () => {
```

Comment:
Consider moving function to be a (private) method of `StorageConfig` to avoid having to use `no-underscore-dangle`.

Reply:
Fixed